src/runtime_src/core/common/api/xrt_bo.cpp

```diff
@@ -729,83 +730,104 @@ namespace xrt {
 
 bo::
 bo(xclDeviceHandle dhdl, void* userptr, size_t sz, bo::flags flags, memory_group grp)
-  : handle(alloc(dhdl, userptr, sz, static_cast<xrtBufferFlags>(flags), grp))
+  : handle(xdpnative::profiling_wrapper(__func__, "xrt::bo",
+           alloc_userptr, dhdl, userptr, sz, static_cast<xrtBufferFlags>(flags), grp))
 {}
 
 bo::
 bo(xclDeviceHandle dhdl, size_t size, bo::flags flags, memory_group grp)
-  : handle(alloc(dhdl, size, static_cast<xrtBufferFlags>(flags), grp))
+  : handle(xdpnative::profiling_wrapper(__func__, "xrt::bo",
+           alloc, dhdl, size, static_cast<xrtBufferFlags>(flags), grp))
 {}
 
 bo::
 bo(xclDeviceHandle dhdl, xclBufferExportHandle ehdl)
-  : handle(alloc(dhdl, ehdl))
+  : handle(xdpnative::profiling_wrapper(__func__, "xrt::bo",
+	   alloc_import, dhdl, ehdl))
 {}
 
 bo::
 bo(const bo& parent, size_t size, size_t offset)
-  : handle(sub_buffer(parent.handle, size, offset))
+  : handle(xdpnative::profiling_wrapper(__func__, "xrt::bo",
+	   sub_buffer, parent.handle, size, offset))
 {}
 
 bo::
 bo(xrtBufferHandle xhdl)
-  : handle(get_boh(xhdl))
+  : handle(xdpnative::profiling_wrapper(__func__, "xrt::bo",
+	   get_boh, xhdl))
 {}
 
 size_t
 bo::
 size() const
 {
-  return handle->get_size();
+  return
+    xdpnative::profiling_wrapper_member(__func__, "xrt::bo",
+                                        handle.get(), &bo_impl::get_size);
```

Comment:
Is profiling_wrapper_member really needed? 
`xdpnative::profiling_wrapper(__func__, "xrt::bo", [this] () { return handle->get_size(); });`

Reply:
wrapper_member function removed and replaced with calls to profiling_wrapper passing lambdas